### PR TITLE
Porvision agent from gallery in different subscription

### PR DESCRIPTION
**Pull request by gavinfish**

### CODE REVIEW

**Comment by gavinfish** [comment]
> What if user doesn't set the subscription id or set an invalid one?

If user does not set the id, it will use the default id of the service principal. No check for invalid one yet.

File: src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-gallerySubscriptionId.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div>
+    The subscription id of the target gallery image belonged to. Leave empty if it is in the same subscription as the Azure Service Principal.
```
Comment (yungezz) [inline src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-gallerySubscriptionId.html]: service principal doesn't has subscription_id, but role assigned to some subscription. how about "same subscription" as configured at  xxxxx

Comment (gavinfish) [reply src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-gallerySubscriptionId.html]: Updated

File: src/main/java/com/microsoft/azure/vmagent/util/AzureClientUtil.java [hunk 2]
```diff
@@ -41,6 +43,18 @@ public static Azure getClient(String credentialId) {
         return getClient(token);
     }
 
+    public static Azure getClient(String credentialId, String subscriptionId) throws AzureCloudException {
+        boolean validSubscriptionId = AzureUtil.isValidSubscriptionId(credentialId, subscriptionId);
+        if (!validSubscriptionId) {
```
Comment (yuwzho) [inline src/main/java/com/microsoft/azure/vmagent/util/AzureClientUtil.java]: is the empty one valid?

Comment (gavinfish) [reply src/main/java/com/microsoft/azure/vmagent/util/AzureClientUtil.java]: yes

**Pull request merged**